packages/core/test/unit/model/scope.test.ts

```diff
@@ -2,14 +2,12 @@ import assert from 'node:assert';
 import { expect } from 'chai';
 import type { FindOptions } from '@sequelize/core';
 import { DataTypes, Op, Sequelize, col, where, literal } from '@sequelize/core';
-// eslint-disable-next-line import/order
-import { resetSequelizeInstance } from '../../support';
+// eslint-disable-next-line import/order -- sequelize cannot be imported right now since there will be too many typing errors
```

Comment:
not worth it to handle in this PR?

Reply:
I didn't look at it for long but most errors come from the following two attributes;
https://github.com/sequelize/sequelize/blob/5f13224dd420604d3f7f3f2969d38cd8f119e445/packages/core/src/model-typescript.ts#L409-L412
And since the TODO has been there since the initial PR that added model-typescript I assumed that typing it would not be that straightforward.

There is also an issue with the typing of Model.withScope() but that might be an easy fix, haven't looked at that at all.

Reply:
I see

Reply:
Quick inventory of the remaining typing errors in that file;

- `_scope` and `_scopeNames` are untyped, see above.
- `Model.withScope()` (with that also `Model.scope()`) also seems to work for multiple string arguments to represent scopes. Now it only accepts one string or an array of strings. (e.g. `Model.withScope('test','second','third')` seems to work fine). Not sure if TypeScript has something to type this with or that we should remove this functionality and use arrays instead.
- `Model._injectScope`, `Model._conformIncludes` and `Model._normalizeIncludes` are untyped. These do not occur often in the tests so if the above two are fixed we can just add a `@ts-expect-error` there temporarily

Reply:
Do you want to handle it here or should I approve and we'll handle that later?

Reply:
I propose we handle it later